### PR TITLE
Drop import-level sampler compilation

### LINKED ISSUE #1245: mlx_lm.server: seed parameter silently ignored
## `mlx_lm.server`: `seed` parameter silently ignored

Chat-completions endpoint accepts `seed` and routes it through `mx.random.seed`, but output is the same across different seed values for the same prompt. mlx-lm 0.31.3, M-series, reproduced on Qwen2.5 and Llama-3.1 4 bit quants.

## Repro

Start server:

```bash
MODEL_PATH=/path/to/your/mlx/model
mlx_lm.server --model "$MODEL_PATH" --port 1234
```

Send three requests with different seeds (same shell, `$MODEL_PATH` carries over):

```bash
for s in 11111 22222 33333; do
  curl -sS http://127.0.0.1:1234/v1/chat/completions \
    -H 'Content-Type: application/json' \
    -d "{\"model\":\"$MODEL_PATH\",\"temperature\":1,\"max_tokens\":25,\"seed\":$s,\"stream\":false,\"messages\":[{\"role\":\"user\",\"content\":\"Tell me a one-sentence story about a cat.\"}]}" \
  | jq -r '.choices[0].message.content'
done
```
```
bash repro.sh
The tabby cat's reflection in the dark window was the only companion she hadn't driven away that evening.
The tabby cat's reflection in the dark window was the only companion she hadn't driven away that evening.
The tabby cat's reflection in the dark window was the only companion she hadn't driven away that evening.
```

All three responses come back byte-identical.

## Expected

Different seeds should follow different sampling trajectories which should provide different outputs (at temperature > 0).

## My thoughts why

`server.py:1191` reads `seed` from the body into `self.seed`. At `server.py:956-957` it's applied:

```python
if args.seed is not None:
    mx.random.seed(args.seed)
```

That call executes. Sampling lives at `sample_utils.py:267-282`:

```python
@partial(mx.compile, inputs=mx.random.state, outputs=mx.random.state)
def categorical_sampling(logits, temp):
    return mx.random.categorical(logits * (1 / temp))
```

~Hypothesis: `mx.compile` with `inputs=mx.random.state` captures the state object at decoration time, so subsequent `mx.random.seed(...)` calls don't propagate into what the compiled function reads. Effectively, sampling becomes deterministic against whatever state existed at module import.~
### Update

Initial suspicion was that the `mx.compile` decorator captured a stale `mx.random.state` reference. Not actually it: running `mx.random.seed(N) -> mx.random.categorical(...)` directly with three different seeds produces three different draws, both with and without the decorator. So it looks like the bug is somewhere downstream of the seed call but upstream of (or alongside) the actual sampling.
## Also tried

- Removed `seed` from the body, ran the same prompt twice. Identical output. State isn't naturally advancing across requests either.
- Same prompt at temperature 0.5 / 1.5 / 3.0. Outputs differ. Sampling runs, temperature is honored. Seed/state path is broken.
- Different prompts at fixed seed. Outputs differ. Not response-caching.

## Workaround

Vary something in the prompt per request (a hidden tag in the system message works). At least breaks the unintended determinism.

### PR DESCRIPTION
**Pull request by michalk8**
Drop module-level `mx.compile` in `sampling_utils.py`. This fixes `server.py`'s sampling seed.

Fixes #1245
Closes #1331

### CODE REVIEW

**Comment by katlun-lgtm** [comment]
Ran this against the repro from #1675 on an M3 Max (mlx 0.32.0, macOS 27.0), comparing `646aa0a9` to stock main at `cf10f96`. It fixes it, and the output matches the uncompiled reference rather than merely being unfrozen:

```
                                main cf10f96          #1593 646aa0a9
worker-built, worker-drawn    [2,2,2,2,2,2,2,2]    [2,7,24,6,0,7,7,7]
uncompiled control (worker)   [2,7,24,6,0,7,7,7]   [2,7,24,6,0,7,7,7]
```

Worth being precise about what changes, because one case in my repro stays frozen on both revisions and I don't think that's a defect: a sampler built on the main thread and *called* from a worker is still stuck. Moving the compile into `make_sampler()` binds the sampler to whichever thread builds it, which isn't the same as making it thread-portable. That's the right trade here, but it does mean the fix is sufficient because of where the call sites happen to sit — `server.py` runs a single `Thread(target=self._generate)` (`:430`), and both `_make_sampler` calls are on it, `:717` in the batch path and `:905` in `_serve_single`, itself invoked from `_generate` at `:750`. Build thread and draw thread coincide, so the state gets read *and* advanced.

The one thing I'd raise for review is that this now-load-bearing property isn't enforced or written down anywhere. If `_make_sampler` ever moves to the HTTP handler thread — which looks like a harmless refactor, building the sampler where the request is parsed — the bug comes back silently, and the test here wouldn't catch it, since it builds and draws on the same thread. A comment at the two call sites would probably be enough; a test that builds on one thread and draws on another, asserting the documented behaviour, would pin it harder. Happy to send either as a patch if it's useful.

**Comment by zcbenz** [comment]
This had been fixed by https://github.com/ml-explore/mlx/pull/3828 and I'm closing this in favor of https://github.com/ml-explore/mlx-lm/pull/1753.
